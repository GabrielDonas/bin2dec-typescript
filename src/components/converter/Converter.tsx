import classNames from "classnames";
import { useState } from "react";
import binaryToDecimal from "../../helpers/binaryToDecimal";
import checkBinary from "../../helpers/checkBinary";
import styles from "./converter.module.css";

function Converter() {
  const [output, setOutput] = useState("0");
  const errorMessage = "Wrong input, please enter either 0 or 1";

  const classes = classNames(
    { [styles.result]: output !== errorMessage },
    { [styles.wrongInput]: output === errorMessage }
  );

  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
    checkBinary(e.target.value)
      ? setOutput(binaryToDecimal(e.target.value))
      : setOutput(errorMessage);
  }

  return (
    <div className={styles.container}>
      <input type="number" name="binary" onChange={handleChange} />
      <div className={classes}>{output}</div>
    </div>
  );
}

export default Converter;
