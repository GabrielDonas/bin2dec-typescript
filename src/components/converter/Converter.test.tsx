import { shallow } from "enzyme";

import Converter from "./Converter";

describe("The Converter component", () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<Converter />);
  });

  it("Should receive only numbers", () => {
    wrapper.instance().handleChange("foo");
    expect(wrapper.state()).toEqual("foo");
  });
});

//configure an Adapter, look at the Terminal
