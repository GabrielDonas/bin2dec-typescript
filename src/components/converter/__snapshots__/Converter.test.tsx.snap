// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Converter component Should render correctly 1`] = `
<div
  className="container"
>
  <input
    name="binary"
    onChange={[Function]}
    type="number"
  />
  <div
    className="result"
  >
    0
  </div>
</div>
`;
